app/models/miq_expression/field.rb

```diff
@@ -2,7 +2,7 @@ class MiqExpression::Field
   FIELD_REGEX = /
 (?<model_name>([[:upper:]][[:alnum:]]*(::)?)+)
 \.?(?<associations>[a-z_\.]+)*
--(?<column>[a-z]+(_[a-z]+)*)
+-(?<column>[[a-z]]+(_[[:alnum:]]+)*)
```

Comment:
- extra brackets: yes there was a typo.

- :alnum: - yes there can be uppercase there as well. Basically custom_attributes is created by name in table CustomAttribute#name and name is string here so it can here basically whatever.

I probably should create some limitation for special characters in follow up PR (one issue of this type was created https://github.com/ManageIQ/manageiq/issues/10482 ) and convert custom attribute column only to form with lower case, upper case, number and underscore as replacement of special characters. But ok let's assume only lower case, upper case, number and underscore here.

so I am not sure if I should can update regex in this way or whether will be better include somehow prefix(`virtual_custom_attribute_`) into regex ? (if possible)